ledger-tool/src/bigtable.rs

```diff
@@ -178,6 +183,141 @@ async fn entries(
     Ok(())
 }
 
+async fn shreds(
+    blockstore: Arc<Blockstore>,
+    starting_slot: Slot,
+    ending_slot: Slot,
+    allow_mock_poh: bool,
+    config: solana_storage_bigtable::LedgerStorageConfig,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let bigtable = solana_storage_bigtable::LedgerStorage::new_with_config(config)
+        .await
+        .map_err(|err| format!("Failed to connect to storage: {err:?}"))?;
+
+    // Make the range inclusive of both starting and ending slot
+    let limit = (ending_slot - starting_slot + 1) as usize;
+    let mut slots = bigtable.get_confirmed_blocks(starting_slot, limit).await?;
+    slots.retain(|&slot| slot <= ending_slot);
+
+    let keypair = Keypair::from_bytes(&[0; 64])?;
+    // TODO: parse this from CLI ?
+    let shred_version = 0;
+    // TODO: parse from CLI OR extract from genesis
+    let num_ticks_per_slot = 64;
+    // TODO: parse from CLI OR extract from Bank; tick rate changed recently
+    let num_hashes_per_tick = 12500;
```

Comment:
Hi @CriesofCarrots - wanted to get some initial thoughts from you on these TODOs. Here is my current thinking:
- `shred_version`: Make this an optional CLI flag; this isn't super critical IMO so fine to leave as optional
    - Also no place that we can fetch this from (don't think we want to introduce RPC calls into ledger-tool)
- `num_ticks_per_slot`: This is currently a fixed value that is available from genesis or a `Bank`
- `num_hashes_per_tick`: Until very recently, this was a fixed value that could be read from genesis. However, this value can now vary with slot so it must be determined from `Bank`

For `num_ticks_per_slot` and `num_hashes_per_tick`, I see two options: 1) require on CLI or 2) read from a `Bank`. 1) would be quicker, but more error prone. 2) will be more correct if the bank is in the same epoch as the desired slot range, but will take more time to execute as the snapshot will have to be unpacked

I'm leaning towards doing 2) so as not to introduce a foot-gun, but curious for a quick sanity check from you as well.

My thinking is that we'd extract a `Bank` from snapshot and use `Bank` helpers to confirm that the desired slot range to create shreds for is the same epoch as the `Bank`'s slot. It could be nice to do this check before unpacking the snapshot to avoid wasted time, but we would need to re-impl some logic to determine epoch from slot ... maybe this wouldn't be so bad

Reply:
I lean toward extracting those two fields from a `Bank` as well. The unfortunate bit is, I guess we would have to error if the `starting_slot..ending_slot` range extends outside that one epoch, or only shred part of the range.

> `shred_version`... Also no place that we can fetch this from (don't think we want to introduce RPC calls into ledger-tool)

Definitely don't want to have to depend on a running node for anything. If we put more requirements on the snapshot being used (or make greater assumptions), I guess we could actually compute the shred_version from the hard_forks in the Bank, right? Not sure it's worth it, though.

Reply:
>  lean toward extracting those two fields from a Bank as well

Cool, this seems like the correct answer, but I was feeling a little lazy so wanted a heat check. Let's get it from the bank.

>  I guess we could actually compute the shred_version from the hard_forks in the Bank, right? Not sure it's worth it, though

Ohh, you might be right! If we can get it from `Bank` easily, then I'm good with doing it that way. I bet we may have a helper for that somewhere already